experiments/findUncoveredAPI.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 The GraphicsFuzz Project Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys, json, subprocess
+
+json_string = subprocess.check_output(
+        "pushd swiftshader/build-coverage/src/Vulkan/CMakeFiles/vk_swiftshader.dir;\
+        gcov -i -t libVulkan.cpp.o;\
+        popd;\
+        exit 0",
+        shell=True,
+        executable="/bin/bash"
+).decode("utf-8").split('\n')[1]
```

Comment:
I suggest not using `shell=True`. Instead, use the `cwd=` parameter to give the directory, and give the command using a list, like so:

```py
[shutil.which("gcov"), "-i", "-t", "libVulkan.cpp.o"]
```

I don't think you need to split the string; you can just use the entire string. Of course, this will fail if there are multiple JSON objects, but if there are multiple objects then something has gone wrong. You could add a comment to note that the output is one JSON object per line, but we only expect one object.

Reply:
I'll make the change to using `cwd=`.

The reason I need to move into the specified directory is because gcov expects that the .gcda file will be in the current working directory. If we pass in a relative directory path instead of just a file name, it would complain that it cannot open the notes file.

The reason I was splitting the string was that pushd and popd emit the current state of their stack to standard output. This would be included in json_string, which doesn't make any sense. So I needed to split the json_string in order to get a valid JSON string.